src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameCheckEnhancedInstanceof.java

```diff
@@ -3,25 +3,29 @@
 
 import java.util.ArrayList;
 import java.util.Locale;
+// config:
+// allowedAbbreviationLength="1"
 
 public class InputAbbreviationAsWordInNameCheckEnhancedInstanceof {
 
     public void t(Object o1, Object o2) {
         // Should cause two violations, STRING and INTEGER, capital letter count > allowed
-        if (!(o1 instanceof String STRING) && (o2 instanceof Integer INTEGER)) {}
+        if (!(o1 instanceof String STRING) // violation
+                && (o2 instanceof Integer INTEGER)) {} // violation
 
         ArrayList<Integer> arrayList = new ArrayList<Integer>();
-        if (arrayList instanceof ArrayList<Integer> aXML) { // ok when allowed "XML" abbreviation
+        if (arrayList instanceof ArrayList<Integer> aXML) { // ok
```

Comment:
`allowedAbbreviationLength="1"` should not allow this

Reply:
XML is an allowed abbreviation in this UT, I've separated the files now.